packages/core/src/Stores/client-store.js

```diff
@@ -763,95 +766,89 @@ export default class ClientStore extends BaseStore {
-            }
-        });
+            localStorage.removeItem('real_account_signup_wizard');
+            this.root_store.gtm.pushDataLayer({ event: 'real_signup' });
+            Promise.resolve({
+                ...response,
+                ...(is_maltainvest_account
+                    ? {
+                          new_account_maltainvest: {
+                              ...response.new_account_maltainvest,
+                              currency,
+                          },
+                      }
+                    : {}),
+            });
+        } else {
+            Promise.reject(response.error);
+        }
     }
 
     @action.bound
-    setAccountCurrency(currency) {
-        return new Promise(async (resolve, reject) => {
-            const response = await WS.setAccountCurrency(currency, {
-                previous_currency: this.currency,
-            });
-            if (!response.error) {
-                runInAction(() => {
-                    const new_account = Object.assign({}, this.accounts[this.loginid]);
-                    new_account.currency = currency;
-                    if (!('balance' in new_account)) new_account.balance = 0;
-                    this.accounts[this.loginid] = new_account;
-                });
-                localStorage.setItem(storage_key, JSON.stringify(this.accounts));
-                LocalStore.setObject(storage_key, JSON.parse(JSON.stringify(this.accounts)));
-                this.selectCurrency(currency);
-                this.root_store.ui.removeNotificationMessage({
-                    key: 'currency',
-                });
-                this.root_store.ui.removeNotificationByKey({
-                    key: 'currency',
-                });
-                await this.init();
-                resolve(response);
-            } else {
-                reject(response.error);
-            }
+    async setAccountCurrency(currency) {
+        const response = await WS.setAccountCurrency(currency, {
+            previous_currency: this.currency,
         });
+        if (!response.error) {
+            runInAction(() => {
+                const new_account = { ...this.accounts[this.loginid] };
+                new_account.currency = currency;
+                if (!('balance' in new_account)) new_account.balance = 0;
+                this.accounts[this.loginid] = new_account;
+            });
+            localStorage.setItem(storage_key, JSON.stringify(this.accounts));
+            LocalStore.setObject(storage_key, JSON.parse(JSON.stringify(this.accounts)));
+            this.selectCurrency(currency);
+            this.root_store.ui.removeNotificationMessage({
+                key: 'currency',
+            });
+            this.root_store.ui.removeNotificationByKey({
+                key: 'currency',
+            });
+            await this.init();
+            Promise.resolve(response);
+        } else {
+            Promise.reject(response.error);
```

Comment:
Same here, while Promises here are created, they aren't returned. (this isn't Ruby 😛 )